scripts/validation.py

```diff
@@ -0,0 +1,143 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Tue May 15 19:30:13 2018
+
+@author: Luzian
+
+USAGE:
+    1st Argument: directory of controlPoints_*.csv
+    2nd Argument: directory of pic*.png, params*.csv files
+    3rd Argument: directory of output images
+"""
+import sys,os,csv,cv2
+import numpy as np
+
+CAMCHAIN_MAT_DIM = (4,4)
+N_STREAMS = 4
+X_SHIFT = 22
+Y_SHIFT = -8
+PREVIEW = 0
+
+def read_matrix_chain(csv_path):
+    csv_path = os.path.abspath(csv_path)
+    with open(csv_path) as csv_file:
+        csv_reader = csv.reader(csv_file)
+        matrices = []
+        for matrix in csv_reader:
+            matrices.append(np.asarray(np.reshape(matrix,CAMCHAIN_MAT_DIM).T,dtype='float64'))# Transpose because MATLAB reshape is column wise and numpy is row-wise
+        return matrices
+
+def read_params(csv_path):
+    csv_path = os.path.abspath(csv_path)
+    params = []
+    with open(csv_path) as csv_file:
+        csv_reader = csv.reader(csv_file)
+        for entry in csv_reader:
+            params.append(entry)
+        return params
+
+def read_controlpoints(csv_path):
+    csv_path = os.path.abspath(csv_path)
+    points = []
+    with open(csv_path) as csv_file:
+        csv_reader = csv.reader(csv_file)
+        for cp in csv_reader:
+            points.append(np.asarray([cp],dtype='float64').T)
+    return points
+
+def make_projection_matrix(camchain,R,cam):
+    matrix = np.identity(4)
+    for i in range(2*cam + 1):
+        matrix = np.matmul(camchain[i],matrix)
+    matrix = np.matmul(R[cam],matrix)
+    return np.asarray(matrix,dtype='float64')
+
+def project_point(control_point, projection_matrix, params):
+    focallength = float(params[0])
+    ccx = float(params[2])
+    ccy = float(params[3])
+    cp_hom = np.row_stack((control_point,[[1]]))
+    p4D = np.matmul(projection_matrix , cp_hom)
+
+    p2D = np.array([p4D[0]/p4D[2]*focallength, p4D[1]/p4D[2]*focallength])
```

Comment:
This line fails, if a controlPoint at (0,0,0) is passed. Maybe ignore them further up? Or should a clean controlPoints.csv never contain a point at (0,0,0)? since the value doesn't make sense, I use it as a placeholder to show, this point should be ignored. We might need to adjust the main exectuable.